File: src/main/java/at/searles/regexp/RepMin.java
package at.searles.regexp;

class RepMin implements Regexp {

    private final Regexp regexp;
    private final int min;

    RepMin(Regexp regexp, int min) {
        this.regexp = regexp;
        this.min = min;
    }

    @Override
    public <A> A accept(Visitor<A> visitor) {
        return visitor.visitRepMin(regexp, min);
    }

    @Override
    public String toString() {
        return String.format("min(%s, %d)", regexp, min);
    }
}

File: src/main/java/at/searles/regexp/RepMin.kt
package at.searles.regexp

internal class RepMin(private val regexp: Regexp, private val min: Int) : Regexp {
    override fun <A> accept(visitor: Visitor<A>): A {
        return visitor.visitRepMin(regexp, min)
    }

    override fun toString(): String {
        return String.format("min(%s, %d)", regexp, min)
    }

}